### PR TITLE
mupdf

### PR DESCRIPTION
**Pull request by scopatz**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by scopatz** [comment]
@conda-forge/staged-recipes @mingwandroid - any hints on why the compilers on Windows wouldn't be able to find an include directory?

File: recipes/mupdf/bld.bat [hunk 1]
```diff
@@ -0,0 +1,18 @@
+@echo off
+setlocal EnableDelayedExpansion
+
+:: build system uses non-standard env vars
+set "XCFLAGS=%CFLAGS% -I%LIBRARY_PREFIX%/include"
```
Comment (xhochy) [inline recipes/mupdf/bld.bat]: Maybe add quotes around `%LIBRARY_PREFIX%/include`?

**Comment by mingwandroid** [comment]
Try removing the double quotes? Set doesn't need them and they are usually problematic.

**Comment by scopatz** [comment]
OK I have tried moving a to cmake build since I need the shared objects anyway.  I don't understand the error though.  Does anyone recognize it?

**Comment by scopatz** [comment]
It just seems to die

**Comment by scopatz** [comment]
CC @conda-forge/staged-recipes - This is ready for review.  I am skipping windows for now, but will fix it up in the feedstock

File: recipes/mupdf/build.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+set -ex
+
+# build system uses non-standard env vars
+uname=$(uname)
+if [[ "$uname" == "Darwin" ]]; then
```
Comment (isuruf) [inline recipes/mupdf/build.sh:6]:
```suggestion
if [[ "$target_platform" == osx* ]]; then
```

File: recipes/mupdf/Makefile-shared-libs-osx.patch [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+-THREAD_LIB = $(OUT)/libmupdf-threads.a
+-PKCS7_LIB = $(OUT)/libmupdf-pkcs7.a
++MUPDF_LIB = $(OUT)/libmupdf.dylib
++THIRD_LIB = $(OUT)/libmupdf-third.dylib
++THREAD_LIB = $(OUT)/libmupdf-threads.dylib
++PKCS7_LIB = $(OUT)/libmupdf-pkcs7.dylib
+ 
+-$(MUPDF_LIB) : $(MUPDF_OBJ)
++$(MUPDF_LIB) : $(MUPDF_OBJ) $(THIRD_LIB) $(THREAD_LIB)
++	$(LINK_CMD) -shared -Wl,-install_name -Wl,libmupdf.dylib -undefined dynamic_lookup $(THIRD_LIBS)
```
Comment (isuruf) [inline recipes/mupdf/Makefile-shared-libs-osx.patch:27]: Does this use `LDFLAGS`?

File: recipes/mupdf/Makerules.patch [hunk 4]
```diff
@@ -0,0 +1,11 @@
+--- Makerules.orig	2020-03-26 15:39:59.725834877 -0500
++++ Makerules	2020-03-26 15:40:10.141808906 -0500
+@@ -62,7 +62,7 @@
+ endif
+ 
+ # Default system libraries
+-SYS_FREETYPE_LIBS := -lfreetype2
++SYS_FREETYPE_LIBS := -lfreetype
```
Comment (isuruf) [inline recipes/mupdf/Makerules.patch:8]: You can do `make SYS_FREETYPE_LIBS="-lfreetype"` and remove this patch

File: recipes/mupdf/Makefile-shared-libs-osx.patch [hunk 5]
```diff
@@ -0,0 +1,39 @@
+--- Makefile.orig	2020-03-25 11:02:34.874709800 -0500
++++ Makefile	2020-04-07 09:39:20.576540717 -0500
+@@ -20,7 +20,7 @@
+ # Do not specify CFLAGS or LIBS on the make invocation line - specify
+ # XCFLAGS or XLIBS instead. Make ignores any lines in the makefile that
+ # set a variable that was set on the command line.
+-CFLAGS += $(XCFLAGS) -Iinclude
++CFLAGS += $(XCFLAGS) -Iinclude -fPIC
```
Comment (isuruf) [inline recipes/mupdf/Makefile-shared-libs-osx.patch:8]: Add `-fPIC` to `XCFLAGS` ?

Comment (scopatz) [reply recipes/mupdf/Makefile-shared-libs-osx.patch:8]: This is part of an upstream patch, so I don't want to change it.

File: recipes/mupdf/Makefile-shared-libs-osx.patch (hunk 3 above)
Comment (scopatz) [reply recipes/mupdf/Makefile-shared-libs-osx.patch:27]: Again, this is part of an upstream patch

File: recipes/mupdf/Makerules.patch (hunk 4 above)
Comment (scopatz) [reply recipes/mupdf/Makerules.patch:8]: That would involve having separate make lines for linux and mac and I thought a small patch was cleaner

**Comment by scopatz** [comment]
OK this is ready for review again

File: recipes/mupdf/Makefile-shared-libs-osx.patch (hunk 3 above)
Comment (isuruf) [reply recipes/mupdf/Makefile-shared-libs-osx.patch:27]: Sure, but is `LDFLAGS` used at all?

**Comment by scopatz** [comment]
CC @conda-forge/staged-recipes

File: recipes/mupdf/Makefile-shared-libs-osx.patch (hunk 3 above)
Comment (scopatz) [reply recipes/mupdf/Makefile-shared-libs-osx.patch:27]: Yes, in the Makefile, there is:

```
LINK_CMD = $(QUIET_LINK) $(MKTGTDIR) ; $(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
```

**Comment by CJ-Wright** [comment]
Good luck with the windows!

**Pull request merged**